cdc/log.hh

```diff
@@ -51,6 +52,26 @@ class database;
 
 namespace cdc {
 
+// cdc log table operation
+enum class operation : int8_t {
+    // note: these values will eventually be read by a third party, probably not privvy to this
+    // enum decl, so don't change the constant values (or the datatype).
+    pre_image = 0, update = 1, insert = 2, row_delete = 3, partition_delete = 4,
+    range_delete_start_inclusive = 5, range_delete_start_exclusive = 6, range_delete_end_inclusive = 7, range_delete_end_exclusive = 8,
+    post_image = 9,
+};
+
+struct per_request_options {
+    // Supports only insert and update for now.
+    std::optional<cdc::operation> log_operation_type_fixup;
+    // Don't generate rows in the log table for this mutation.
+    bool skip_cdc;
+    // The value of the base row before current operation, queried by higher
+    // layers than CDC. We assume that CDC could have seen the row in this
+    // state, i.e. the value isn't 'stale'/'too new'.
+    lw_shared_ptr<cql3::untyped_result_set> preimage;
```

Comment:
Make sure that schema is correct (i.e. cdc's preimage may skip some columns in the select?)